codecparsers/jpegParser.cpp

```diff
@@ -313,6 +313,9 @@ bool Parser::nextMarker()
     const uint8_t* current = start;
     const uint8_t* match;
 
```

Comment:
OK, I see now... `currentBytePosition() < m_size` is a bad assumption.

So all we need is
```
if (m_input.getRemainingBitsCount() == 0)
  return false;
```
at top of function.